### PR TITLE
Ignore some plugins. Add few more known plugin types

### PR DESCRIPTION
**Pull request by dbuzhorkaa**

### CODE REVIEW

File: scripts/generate-plugins/main.py [hunk 1]
```diff
@@ -46,7 +49,7 @@
   'windower': 'Analytics',
   'batchjoiner': 'Analytics',
   'condition': 'Condition',
-  'errortransform': 'Error Handler'
+  'errortransform': 'Error Handler',
```
Comment (bdmogal) [inline scripts/generate-plugins/main.py]: is comma at the end supported in python?

File: .pluginsignore [hunk 2]
```diff
@@ -1 +1,6 @@
 # Specify plugin names to ignore
+KAFKABATCHSOURCE
```
Comment (bdmogal) [inline .pluginsignore]: Should we move this file under scripts/plugin..

File: scripts/generate-plugins/main.py [hunk 3]
```diff
@@ -30,14 +30,17 @@
 
 
 PLUGIN_DISPLAY_TYPES = {
+  'source': 'Source',
```
Comment (bdmogal) [inline scripts/generate-plugins/main.py]: these shouldn't be needed. Where do you see these? Only batchsource and streamingsource are supported source types in CDAP.

File: scripts/generate-plugins/main.py [hunk 4]
```diff
@@ -30,14 +30,17 @@
...
   'batchsource': 'Source',
   'streamingsource': 'Source',
+  'sink': 'Sink',
   'batchsink': 'Sink',
   'sparksink': 'Sink',
   'realtimesink': 'Sink',
   'transform': 'Transform',
   'splittertransform': 'Transform',
   'alertpublisher': 'Alert Publisher',
+  'alert': 'Alert Publisher',
```
Comment (bdmogal) [inline scripts/generate-plugins/main.py]: ditto

File: scripts/generate-plugins/main.py [hunk 5]
```diff
@@ -30,14 +30,17 @@
 
 
 PLUGIN_DISPLAY_TYPES = {
+  'source': 'Source',
   'batchsource': 'Source',
   'streamingsource': 'Source',
+  'sink': 'Sink',
```
Comment (bdmogal) [inline scripts/generate-plugins/main.py]: ditto

File: scripts/generate-plugins/main.py (hunk 3 above)
Comment (dbuzhorkaa) [reply scripts/generate-plugins/main.py]: I got this value during plugin generation as `plugin_type`. Thought that this was a valid type

File: scripts/generate-plugins/main.py (hunk 4 above)
Comment (dbuzhorkaa) [reply scripts/generate-plugins/main.py]: Please see my comment above

File: .pluginsignore (hunk 2 above)
Comment (dbuzhorkaa) [reply .pluginsignore]: I can move it. My initial thought was to hold this file at the root level directory along with site configuration files. Let me know if I need to move it

Comment (bdmogal) [reply .pluginsignore]: yeah I think let's move it, since it only applies to a singe script.

File: scripts/generate-plugins/main.py (hunk 3 above)
Comment (bdmogal) [reply scripts/generate-plugins/main.py]: Could you tell me which plugins emit these as types?

Comment (dbuzhorkaa) [reply scripts/generate-plugins/main.py]: Kudu and Kafka

File: .pluginsignore (hunk 2 above)
Comment (dbuzhorkaa) [reply .pluginsignore]: done

File: scripts/generate-plugins/main.py (hunk 3 above)
Comment (bdmogal) [reply scripts/generate-plugins/main.py]: Are these the ones?
image


If so, once you add them to the excludes list, you shouldn't see them, right?

Comment (dbuzhorkaa) [reply scripts/generate-plugins/main.py]: Yes, correct

Comment (dbuzhorkaa) [reply scripts/generate-plugins/main.py]: Do I need to exclude them?

Comment (bdmogal) [reply scripts/generate-plugins/main.py]: yes, all these should be excluded, and so you won't need any of the changes to that plugin types dictionary above.

File: scripts/generate-plugins/main.py (hunk 1 above)
Comment (dbuzhorkaa) [reply scripts/generate-plugins/main.py]: done

File: scripts/generate-plugins/main.py (hunk 3 above)
Comment (dbuzhorkaa) [reply scripts/generate-plugins/main.py]: done

File: scripts/generate-plugins/main.py (hunk 5 above)
Comment (bdmogal) [reply scripts/generate-plugins/main.py]: this can be removed, right?

Comment (dbuzhorkaa) [reply scripts/generate-plugins/main.py]: done

**Comment by dbuzhorkaa** [comment]
@bdmogal I also added `KAFKAWRITER` to excluded list because it type was related to one that I removed (`sink`). Is this ok?

**Comment by bdmogal** [comment]
Yup. That's good.

**Pull request merged**